src/main/java/net/java/otr4j/session/Session.java

```diff
@@ -619,13 +609,28 @@ private String handleDataMessage(DataMessage data) throws OtrException {
                 if (mostRecent.getRemoteKeyID() == senderKeyID)
                     this.rotateRemoteSessionKeys(data.nextDH);
 
-                // Handle TLVs
+                // find the null TLV separator in the package, or just use the end value
+                int tlvIndex = dmc.length;
+                for (int i = 0; i < dmc.length; i++) {
+                    if (dmc[i] == 0x00) {
+                        tlvIndex = i;
+                        break;
+                    }
+                }
+
+                // get the message body without the trailing 0x00, expect UTF-8 bytes
+                String decryptedMsgContent;
+                try {
+                    decryptedMsgContent = new String(dmc, 0, tlvIndex, "UTF-8");
```

Comment:
If you use SerializationUtils.UTF8 here, you do not need to catch UnsupportedEncodingException. Makes the whole thing more readable.